consensus/scp/src/slot.rs

```diff
@@ -526,11 +526,8 @@ impl<V: Value, ValidationError: Display> Slot<V, ValidationError> {
         // Invariant: X and Y are disjoint.
         assert!(self.X.is_disjoint(&self.Y));
 
-        let mut new_Z = self.additional_values_confirmed_nominated();
-        if !new_Z.is_empty() {
-            new_Z.append(&mut self.Z);
-            self.Z = (self.combine_fn)(new_Z);
```

Comment:
Previously, Z always held a set of values that had been through the combine function. This had the advantage that Z could be copied into a Ballot at any time, but had the downside that a node might "forget" that a value had been confirmed nominated if that value were excluded by the combine function. I think this situation is benign, but might mean repeatedly checking if a value has been accepted nominated by a quorum.

In this PR, Z contains all values that have been confirmed nominated, and the combine function is only applied when the values in Z are moved into a Ballot. I think this is logically clearer, and might be slightly more efficient.